otherlibs/stdune/src/user_message.ml

```diff
@@ -226,6 +227,12 @@ let pp { loc; paragraphs; hints; annots = _ } =
            (Pp.textf "File %S, %s, characters %d-%d:" start.pos_fname lnum start_c stop_c))
       :: paragraphs
   in
+  let paragraphs =
+    match context with
+    | None | Some "default" | Some ".sandbox" -> paragraphs
```

Comment:
If you can pass a `Context_name.t`, you could use `Context_name.is_default` here. There are other non-context directories in `_build`, like `.actions`, not sure if they need special handling.

Reply:
I don't think I can use `Context_name` in `User_message`, because the latter exists in stdune and the former in dune_engine. Unless we move `Context_name` from dune_engine to stdune, but I am assuming if the plan is to get rid of `Context_name` in [the engine](https://github.com/ocaml/dune/pull/10414#issuecomment-2050294373), then it doesn't make sense to keep it in stdune? I'm not sure i'm understanding the whole thing.

Reply:
Ah, sorry I missed this.
Maybe it could work to define a type like `No_context | Default | Other_context of string` and use it here? But then it means that `dune_util` needs to parse the context name.

Reply:
Sorry, I am not sure I understand. `dune_util` depends on `stdune`. If we define a type in `dune_util`, how could it be used in `stdune` without getting cycles? Is the suggestion to move `User_message` over to `dune_util`?

> Maybe it could work to define a type like

I am defining a similar type in the related PR #10324: 

https://github.com/ocaml/dune/blob/6c2baf925e9bd07377443aecac1c540f8e756b0f/bin/ocaml/ocaml_merlin.ml#L14-L16

Reply:
ok, nevermind the suggestion then. this is not a hard blocker for me